Interni/NormeProgetto/components/norme-c3.tex

```diff
@@ -0,0 +1,475 @@
+\subsection{Processo di verifica}
+\subsubsection{Scopo}
+Il processo di verifica si occupa di determinare se i prodotti di una data attività sono corretti, coesi e completi e di individuare eventuali errori introdotti nella fase di sviluppo. Sono soggetti a verifica sia il software che la documentazione.
+\subsubsection{Descrizione}
+Il processo di verifica si compone di due attività che permettono di verificare che il prodotto sia conforme alle aspettative di analisi (\S{}\ref{Analisi}) e di test (\S{}\ref{Test}). Esso viene applicato su ogni processo in esecuzione:
+\begin{itemize}
+	\item al raggiungimento di un livello di maturità sufficiente;
+	\item successivamente a cambiamenti inerenti lo stato.
+\end{itemize}
+\subsubsection{Aspettative}
+Il processo di verifica si occupa di prendere ciò che è stato prodotto e di restituirlo in uno stato conforme alle aspettative. Per assicurare il corretto funzionamento di tale processo devono essere rispettate alcune regole:
+\begin{itemize}
+	\item seguire procedure definite;
+	\item seguire criteri chiari ed affidabili;
+	\item i prodotti devono passare attraverso fasi successive opportunamente verificate;
+	\item il prodotto deve trovarsi in uno stato stabile dopo la verifica;
+	\item alla fine del processo deve essere possibile procedere alla fase di validazione.
+\end{itemize}
+\subsubsection{Attività}
+\paragraph{Analisi}
+\label{Analisi}
+Il processo di analisi si suddivide in Analisi statica ed Analisi dinamica.
+\subparagraph{Analisi statica}
+L'analisi statica permette di effettuare controlli su documenti e codice, verificando così che la corretta applicazione della correttezza (intesa come assenza di errori e difetti). Vi sono due metodologie per applicarla:
+\begin{itemize}
+	\item \textbf{Walkthrough:} tecnica di verifica consistente nella lettura da parte del team dell'intero documento o codice in cerca di anomalie. \textit{Verrà applicata principalmente nella fase iniziale del progetto, quando il team ancora non conosce i possibili errori che si possono incontrare}. Questa tecnica risulta molto onerosa in termini di efficienza ed efficacia.
+	\item \textbf{Inspection:} tecnica di verifica attraverso il quale il \textit{Verificatore} eseguirà una lettura mirata e strutturata del documento o del codice nei punti in cui si sa già che possano essere presenti degli errori. Risulta meno dispendiosa in termini di tempo ma richiede una buona conoscenza di fondo.
+\end{itemize}
+\subparagraph{Analisi dinamica}
+L'analisi dinamica prevede l'esecuzione del prodotto software e la sua analisi tramite l'utilizzo di test che verificano se il prodotto funziona o se vi sono presenti anomalie.
+\paragraph{Test}
+\label{Test}
+I test rappresentano l'attività sulla quale si basa l'analisi dinamica. Essi permettono di individuare tutti i possibili errori che possono essere stati commessi e tutti i casi limite che possono risultare problematici. Per ogni test devono essere definiti i seguenti parametri:
```

Comment:
cambiare 
"I test rappresentano l'attività sulla quale si basa l'analisi dinamica."
in
"L'attività di testing è la base dell'analisi dinamica."